lab_5_scraper/scraper.py

```diff
@@ -129,11 +243,30 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
+        all_links = article_bs.find_all('a', href=True)
+        for link in all_links:
+            href = link['href']
+            if href.startswith('/news/'):
+                full_url = 'https://ks-yanao.ru' + href
+                if full_url not in self.urls:
```

Comment:
this function is not `_exctract_url_if_it_is_not_in_self_urls`, move this check outside